watson_developer_cloud/speech_to_text_v1.py

```diff
@@ -18,10 +18,79 @@
 
 from .watson_developer_cloud_service import WatsonDeveloperCloudService
 import json
-
+import base64
+
+from autobahn.twisted.websocket import WebSocketClientFactory,\
+    WebSocketClientProtocol, connectWS
+from twisted.internet import ssl, reactor
+
+
+class STTWebsocketClientProtocol(WebSocketClientProtocol):
+    def setFile(self, filename):
+        self.filename = filename
+    def setContentType(self, content_type):
+        self.contentType = content_type
+    def setFactory(self, factory):
+        self.factory = factory
+    def setCallback(self, callableobj):
+        self.messageCallback = callableobj
+    def setParams(self, params):
+        self.params = params
+    def onOpen(self):
+        data = self.params
+        # send the initialization parameters
+        self.sendMessage(json.dumps(data).encode('utf-8'))
+        with open(self.filename, 'rb') as audiofile:
+            while True:
+                buf = audiofile.read(1024)
+                if not buf:
+                    self.sendMessage(json.dumps({'action': 'stop'}).encode('utf-8'))
+                    break
+                self.sendMessage(buf, isBinary=True)
+
+    def onMessage(self, payload, is_binary):
+        if not is_binary:
+            json_payload = json.loads(payload)
+            try:
+                self.messageCallback(json_payload)
+            except AttributeError:
+                # if we don't have a message callback, don't worry about it
+                pass
+            if 'results' in json_payload:
+                if len(json_payload['results']) > 0:
+                    if 'final' in json_payload['results'][0]:
+                        if json_payload['results'][0]['final']:
```

Comment:
I don't think we want to close the `WebSocket` when`final=true`. The user can send more audio and final just means that the server won't go back to the sentence and change it.